index.js

```diff
@@ -75,119 +71,114 @@ var PersonaClient = function (config) {
+        };
 
-            if (usingScope != null) {
-                requestPath += "?scope=" + usingScope;
+        jwt.verify(token, _this.publicKey, jwtConfig, function onVerify(error, decodedToken) {
+            if(error) {
+                return next(ERROR_TYPES.VALIDATION_FAILURE, null);
             }
 
-            var options = {
-                hostname: _this.config.persona_host,
-                port: _this.config.persona_port,
-                path: requestPath,
-                method: 'HEAD'
-            };
-
-            _this.debug(JSON.stringify(options));
-
-            _this.http.request(options, function (oauthResp) {
-                if (oauthResp.statusCode === 204) {
-                    // put this key in redis with an expire
-                    _this.redisClient.multi().set("access_token:" + cacheKey, 'OK').expire("access_token:" + cacheKey, 60).exec(function (err, results) {
-                        _this.debug("cache: " + JSON.stringify(err) + JSON.stringify(results));
-                    });
-                    _this.debug("Verification passed for token " + cacheKey + ", cached for 60s");
-                    next(null, VALIDATED_TYPES.PERSONA);
-                } else {
-                    if (usingScope && usingScope !== "su") {
-                        // try su, they are allowed to perform any operation
-                        _this.validateToken (token, "su", overridingScope, next);
-                    } else {
-                        _this.debug("Verification failed for token " + cacheKey + " with status code " + oauthResp.statusCode);
-                        if (oauthResp.statusCode === 403) {
-                            next(ERROR_TYPES.INSUFFICIENT_SCOPE, null);
-                        } else {
-                            next(ERROR_TYPES.VALIDATION_FAILURE, null);
-                        }
-                    }
-                }
-            }).on ("error", function (e) {
-                _this.error("OAuth::validateToken problem: " + e.message);
-                next(e.message, null);
-            }).end();
-        }
-    });
+            if (_.includes(decodedToken.scopes, "su") || _.intersection(decodedToken.scopes, scopes).length > 0) {
+                return next(null, "ok");
+            } else {
+                return next(ERROR_TYPES.INSUFFICIENT_SCOPE, null);
+            }
+        });
+    }
+
+    if(this.publicKey == null) {
+        var options = {
+            hostname: this.config.persona_host,
+            port: this.config.persona_port,
+            path: '/oauth/keys',
+            method: 'GET'
+        };
+
+        this.http.request(options, function onSuccess(response) {
+            var publicKey = "";
+            response.on("data", function onData(chunk) {
+                publicKey += chunk;
+            });
+            response.on("end", function onEnd() {
+                _this.publicKey = publicKey;
+                verifyToken();
```

Comment:
opinionated comment: add a return before the function call to make it explicit where the end is